src/main/java/com/google/firebase/auth/FirebaseAuth.java

```diff
@@ -605,6 +605,56 @@ protected UserRecord execute() throws FirebaseAuthException {
     };
   }
 
+  /**
+   * Gets the user data corresponding to the specified user\'s federated identifier.
+   *
+   * @param providerUid The user identifier with the given provider.
+   * @param providerId Identifier for the given federated provider, for example,
+   *     "google.com" for the Google provider.
+   * @return A {@link UserRecord} instance.
+   * @throws IllegalArgumentException If the providerUid is null or empty, or if
+   *     the providerId is null, empty, or does not belong to a federated provider.
+   * @throws FirebaseAuthException If an error occurs while retrieving user data.
+   */
+  public UserRecord getUserByFederatedId(
```

Comment:
Should be getUserByProviderUid(). (Though I think originally, this name was correct.) Also below (getUserByProviderUidAsync)